collect_app/src/main/java/org/odk/collect/android/activities/FormEntryActivity.java

```diff
@@ -2282,6 +2283,8 @@ public void denied() {
                 t.destroy();
 
                 Collect.getInstance().setFormController(formController);
+                formControllerAvailable(formController);
```

Comment:
This change makes sense. I wonder if we should also maybe dismiss the loading dialog later once we set everything because now we do that at the beginning of this method what unblocks the UI and if the rest of the job we do in this method takes some time it's dangerous.